share/functions/__fish_make_cache_dir.fish

```diff
@@ -9,6 +9,21 @@ function __fish_make_cache_dir --description "Create and return XDG_CACHE_HOME"
     # So if you call `__fish_make_cache_dir completions`,
     # this creates e.g. ~/.cache/fish/completions
     if not path is -d $xdg_cache_home/fish/"$argv[1]"
-        mkdir -m 700 -p $xdg_cache_home/fish/"$argv[1]"
+        set -l mkdir_options -m 700
+
+        # Can't set the permission in Cygwin on a `noacl` mount
+        if uname | string match -qr "^(MSYS|CYGWIN)"
```

Comment:
Could also use "__fish_uname" instead of "uname", to save on forks,
although this one is probably not called very often anyway.